silkworm/rpc/core/logs_walker.cpp

```diff
@@ -149,45 +149,50 @@ Task<void> LogsWalker::get_logs(BlockNum start,
     Logs filtered_chunk_logs;
 
     uint64_t block_timestamp{0};
+    std::shared_ptr<BlockWithHash> block_with_hash{};
     auto itr = db::txn::make_txn_nums_stream(std::move(paginated_stream), asc_order, tx_, provider);
     while (const auto tnx_nums = co_await itr->next()) {
-        if (tnx_nums->final_txn) {
-            continue;
-        }
+        SILK_DEBUG << " blockNum: " << tnx_nums->block_num << " txn_index: " << tnx_nums->txn_index.value_or(0) << " txn_id: " << tnx_nums->txn_id << " initial: " << tnx_nums->initial_txn << " final: " << tnx_nums->final_txn;
+
         if (tnx_nums->block_changed) {
             receipts.clear();
 
-            const auto block_with_hash = co_await rpc::core::read_block_by_number(block_cache_, *chain_storage, tnx_nums->block_num);
+            block_with_hash = co_await rpc::core::read_block_by_number(block_cache_, *chain_storage, tnx_nums->block_num);
             if (!block_with_hash) {
                 SILK_DEBUG << "Not found block no.  " << tnx_nums->block_num;
                 break;
             }
             block_timestamp = block_with_hash->block.header.timestamp;
-            receipts = co_await core::get_receipts(tx_, *block_with_hash, *chain_storage, workers_);
-            SILK_DEBUG << "Read #" << receipts.size() << " receipts from block " << tnx_nums->block_num;
         }
```

Comment:
I suggest to add an assertion here: `SILKWORM_ASSERT(block_with_hash);`

Reply:
ok